src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -554,5 +555,47 @@ public enum SampleEnumSByte : sbyte
             F = 1 << 5,
             G = 1 << 6,
         }
+
+        [Fact]
+        public static void HonorNamingPolicyOnDeserialization()
+        {
+            JsonSerializerOptions options = new()
+            {
+                Converters =
+                {
+                    new JsonStringEnumConverter(namingPolicy: new SimpleSnakeCasePolicy() )
+                }
+            };
+
+            BindingFlags bindingFlags = JsonSerializer.Deserialize<BindingFlags>(@"""non_public""", options);
+            Assert.Equal(BindingFlags.NonPublic, bindingFlags);
+
+            // Flags supported without naming policy.
+            bindingFlags = JsonSerializer.Deserialize<BindingFlags>(@"""NonPublic, Public""", options);
```

Comment:
Okay. For correctness/completeness we can populate a cache with naming-policy-formatted text for _all_ enum values. This would be consulted to support all enum values, including flags.
As previously mentioned, I worry that a lot of memory could be consumed supporting this feature 🤷, but this is capped by the number of enums in the (de)serialization routine and the number of values for each enum.

Reply:
> I worry that a lot of memory could be consumed supporting this feature 🤷

I think that's fine, 99% of users don't serialize flags but we still need to be consistent with how we apply naming policies across the board, even if that means a few additional allocations (assuming users exceed the cache soft limit).